src/playback-watcher.js

```diff
@@ -100,6 +101,18 @@ export default class PlaybackWatcher {
     }
 
     let currentTime = this.tech_.currentTime();
+    let buffered = this.tech_.buffered();
+
+    if (!buffered.length || currentTime + 0.1 >= buffered.end(buffered.length - 1)) {
```

Comment:
I think that `!buffered.length` will prevent us from checking for `fellOutOfLiveWindow_` and running `fixesBadSeeks_ ` on environments where the `waiting` event might not fire and we rely on our implementation.